pallets/claims/src/lib.rs

```diff
@@ -0,0 +1,207 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+use codec::{Decode, Encode};
+use frame_support::{
+	decl_error, decl_event, decl_module, decl_storage,
+	dispatch::DispatchResult,
+	ensure,
+	traits::{Currency, Get, Imbalance},
+	weights::{DispatchClass, Pays},
+};
+use frame_system::ensure_signed;
+use primitives::Balance;
+use sp_runtime::traits::Zero;
+use sp_std::prelude::*;
+use sp_std::vec::Vec;
+
+use frame_support::sp_runtime::traits::{DispatchInfoOf, SignedExtension};
+use frame_support::sp_runtime::transaction_validity::{
+	InvalidTransaction, TransactionValidity, TransactionValidityError, ValidTransaction,
+};
+use frame_support::traits::IsSubType;
+use sp_std::marker::PhantomData;
+
+pub use traits::*;
+
+mod claims_data;
+mod migration;
+mod traits;
+
+#[cfg(test)]
+mod mock;
+
+#[cfg(test)]
+mod tests;
+
+mod benchmarking;
+
+use weights::WeightInfo;
+
+pub mod weights;
+
+pub trait Config: frame_system::Config {
+	type Event: From<Event<Self>> + Into<<Self as frame_system::Config>::Event>;
+	type Prefix: Get<&'static [u8]>;
+	type WeightInfo: WeightInfo;
+	type Currency: Currency<Self::AccountId>;
+	// This type is needed to convert from Currency to Balance
+	type CurrencyBalance: From<Balance>
+		+ Into<<Self::Currency as Currency<<Self as frame_system::Config>::AccountId>>::Balance>;
+}
+
+pub type BalanceOf<T> = <<T as Config>::Currency as Currency<<T as frame_system::Config>::AccountId>>::Balance;
+
+decl_storage! {
+	trait Store for Module<T: Config> as Claims {
+		Claims get(fn claims): map hasher(blake2_128_concat) EthereumAddress => BalanceOf<T>;
+
+		PalletVersion: StorageVersion = StorageVersion::V1EmptyBalances;
+	}
+
+	add_extra_genesis {
+		config(claims): Vec<(EthereumAddress, BalanceOf<T>)>;
+
+		build(|config: &GenesisConfig<T>| {
+			config.claims.iter().for_each(|(eth_address, initial_balance)| {
+				Claims::<T>::mutate(eth_address, |amount| *amount += *initial_balance)
+			})
+		})
+	}
+}
+
+decl_event!(
+	pub enum Event<T>
+	where
+		AccountId = <T as frame_system::Config>::AccountId,
+		Balance = BalanceOf<T>,
+	{
+		Claimed(AccountId, Balance),
```

Comment:
Events shouldn't be in past tense `Claimed` -> `Claim` and this particular event should contain ETH address from which was this balance claimed.

Reply:
Currently, the same information will be laso emitted in the `Endowed` event so the ETH claim address is the most useful information transmitted by this event.

Reply:
In substrate recipe describing events there is the present tense so we can stick with that.
ETH address added